plugins/spender/splice.c

```diff
@@ -0,0 +1,1402 @@
+		json_array_end(response);
+	}
+	return command_finished(cmd, response);
+}
+
+static struct command_result *execute_splice(struct command *cmd,
+					      struct splice_cmd *splice_cmd)
+{
+	struct splice_script_result *action;
+	struct splice_cmd_action_state *state;
+	struct wally_psbt_output *output;
+	u64 serial_id;
+	int pays_fee;
+	jsmntok_t tok;
+	const u8 *scriptpubkey;
+
+	/* Basic validation */
+	pays_fee = 0;
+	for (size_t i = 0; i < tal_count(splice_cmd->actions); i++) {
+		int dest_count = 0;
+		action = splice_cmd->actions[i];
+		state = splice_cmd->states[i];
+
+		if (splice_cmd->actions[i]->out_ppm)
+			return do_fail(cmd, splice_cmd, JSONRPC2_INVALID_PARAMS,
+					    "Should be no out_ppm on final");
+		if (splice_cmd->actions[i]->pays_fee && pays_fee++)
+			return do_fail(cmd, splice_cmd, JSONRPC2_INVALID_PARAMS,
+					    "Only one item may pay fee");
+		if (splice_cmd->actions[i]->channel_id)
+			dest_count++;
+		if (splice_cmd->actions[i]->bitcoin_address)
+			dest_count++;
+		if (splice_cmd->actions[i]->onchain_wallet)
+			dest_count++;
+		if (dest_count < 1)
+			return do_fail(cmd, splice_cmd, JSONRPC2_INVALID_PARAMS,
+					    "Must specify 1 destination per");
+		if (dest_count > 1)
+			return do_fail(cmd, splice_cmd, JSONRPC2_INVALID_PARAMS,
+					    "Too many destinations per");
+
+		/* If user specifies both sats in and out, we just use the
+		 * larger of the two and subtract the smaller. */
+		if (amount_sat_greater(action->in_sat, action->out_sat)) {
+			if (!amount_sat_sub(&action->in_sat, action->in_sat,
+					    action->out_sat))
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Unable to sub out_sat from"
+						    " in_sat");
+			action->out_sat = amount_sat(0);
+		} else {
+			if (!amount_sat_sub(&action->out_sat, action->out_sat,
+					    action->in_sat))
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Unable to sub in_sat from"
+						    " out_sat");
+			action->in_sat = amount_sat(0);
+		}
+	}
+
+	add_to_debug_log(splice_cmd, "execute_splice");
+
+	for (size_t i = 0; i < tal_count(splice_cmd->actions); i++) {
+		action = splice_cmd->actions[i];
+		state = splice_cmd->states[i];
+
+		/* Load (only one) feerate if user provided one */
+		if (action->feerate_per_kw) {
+			if (splice_cmd->feerate_per_kw)
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Only one item may set"
+						    " feerate");
+			splice_cmd->feerate_per_kw = action->feerate_per_kw;
+		}
+
+		/* Fund out to bitcoin address */
+		if (action->bitcoin_address) {
+			if (!amount_sat_zero(action->in_sat))
+				return do_fail(cmd, splice_cmd,
+						    JSONRPC2_INVALID_PARAMS,
+						    "Cannot fund from bitcoin"
+						    " address");
+			tok.type = JSMN_STRING;
+			tok.start = 0;
+			tok.end = strlen(action->bitcoin_address);
+			tok.size = tok.end;
+			switch(json_to_address_scriptpubkey(cmd,
```

Comment:
this is so hacky. we can have nice things! just make a bitcoin address -> scriptpubkey method?

Reply:
I began the work on making an address -> scriptpubkey function but it's incomplete.

Moving it to a new task: https://github.com/ElementsProject/lightning/issues/5911